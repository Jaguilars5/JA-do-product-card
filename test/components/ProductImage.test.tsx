import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductImage', () => {
  test('Debe usar el url proporcionado', () => {
    const wrapper = renderer.create(<ProductImage img='htpp/google.com' />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('Debe de mostrar en el componente con el url del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
