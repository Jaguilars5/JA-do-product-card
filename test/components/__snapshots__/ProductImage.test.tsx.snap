// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage Debe de mostrar en el componente con el url del producto 1`] = `
<div
  className=" productCard undefined "
>
  <img
    alt="Coffee Mug"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductImage Debe usar el url proporcionado 1`] = `
<img
  alt="Coffee Mug"
  className="productImg undefined"
  src="htpp/google.com"
/>
`;
